lib/booking_manager.rb

```diff
@@ -0,0 +1,59 @@
+require_relative 'reservation.rb'
+require 'pry'
+
+class BookingManager
+
+  attr_reader :rooms
+  attr_accessor :reservations, :start_date, :end_date, :make_reservation
+
+
+  def initialize
+    @start_date = start_date
+    @end_date = end_date
+    @reservations = []
+    @rooms = []
+    i = 0
+    20.times do |i|
+      @rooms << i + 1
+      i += 1
+    end
+
+  end
+
+  def find_available_room(start_date, end_date)
+    requested_dates = (start_date...end_date).to_a
+    room = 237
+    room_count = 1
+    while room_count < 20
+      if @reservations.empty?
+        room = 1
+        return room
+      end
+      @reservations.each_with_index do |reservation, index|
+        if reservation.room == (index + 1) && (reservation.reservation_dates & requested_dates).length == 0
+          room = index + 1
+          return room
+        elsif reservation.room == (index + 1)
+          room_count += 1
+        elsif reservation.room != (index + 1)
+          room = index + 1
+          return room
+        end
+      end
+    end
+    raise StandardError.new("No room available")
+  end
+end
+
+def make_reservation(start_date, end_date)
+  start_date = @start_date
+  end_date = @end_date
```

Comment:
What's going on in these lines?

In line 48, you define a method named `make_reservation`, which takes in two pieces of data. The first parameter becomes the local variable named `start_date`, and the second parameter becomes the local variable named `end_date`.

In line 49, you say that the value of the local variable `start_date` is ACTUALLY reassigned to the value of the instance variable `@start_date` ... which, in most cases of your code running, is `nil`, because it's not defined (correctly) before this.

Similarly, in line 50, you say that the value of the local variable `end_date` is ACTUALLY reassigned to the value of the instance variable `@end_date`, or `nil`.

What were you trying to do?

For me, this says you aren't sure about how to use instance variables and local variables. Reach out to me or Dan if you want to go over it.